src/web/src/components/SearchInput/SearchInput.tsx

```diff
@@ -1,11 +1,23 @@
 import PostSearchInput from './PostSearchInput';
 import AuthorSearchInput from './AuthorSearchInput';
+import DateSearchInput from './DateSearchInput';
 import useSearchValue from '../../hooks/use-search-value';
 
 const SearchInput = () => {
   const { filter } = useSearchValue();
 
-  return filter === 'author' ? <AuthorSearchInput /> : <PostSearchInput />;
+  function getComponent(f: string) {
```

Comment:
Why bother with a function you immediately call?  Just inline the function body.